### ISSUE TITLE
Kubernetes daemonset fails to deploy

### ISSUE THREAD

**Issue by mellotron** [cilium/cilium#669]
The daemonset here: https://github.com/cilium/cilium/blob/master/contrib/packaging/k8s/daemon-set.yaml

Fails to deploy with the following:

```
$ kubectl logs cilium-net-controller-4dmwf
Error: unknown flag: --ipv4
unknown flag: --ipv4
Usage:
  daemon [flags]

Flags:
      --access-log string             Path to access log of all HTTP requests observed
      --bpf-root string               Path to mounted BPF filesystem
      --config string                 config file (default is $HOME/ciliumd.yaml)
      --consul string                 Consul agent address [127.0.0.1:8500]
  -D, --debug                         Enable debug messages
  -d, --device string                 Device to snoop on (default "undefined")
      --disable-conntrack             Disable connection tracking
      --disable-ipv4                  Disable IPv4 mode
  -e, --docker string                 Register a listener for docker events on the given endpoint (default "unix:///var/run/docker.sock")
      --enable-policy                 Enable policy enforcement
      --enable-tracing                Enable tracing while determining policy
      --etcd stringSlice              Etcd agent address [http://127.0.0.1:2379]
      --etcd-config-path string       Absolute path to the etcd configuration file
      --ipv4-range string             IPv4 prefix
      --k8s-api-server string         Kubernetes api address server
      --k8s-kubeconfig-path string    Absolute path to the kubeconfig file
      --k8s-prefix stringSlice        Key values that will be read from kubernetes. (Default: k8s-app, version)
      --keep-config                   When restoring state, keeps containers' configuration in place
      --keep-templates                Do not restore template files from binary
      --kvstore string                Key-value store type (default "local")
      --label-prefix-file string      File with valid label prefixes
      --labels stringSlice            List of label prefixes used to determine identity of an endpoint
      --lb string                     Enables load balancer mode where load balancer bpf program is attached to the given interface
      --lib-dir string                Path to store runtime build environment (default "/var/lib/cilium")
      --logstash                      Enable logstash integration
      --logstash-agent string         Logstash agent address (default "127.0.0.1:8080")
      --logstash-probe-timer uint32   Logstash probe timer (seconds) (default 10)
      --nat46-range string            IPv6 prefix to map IPv4 addresses to (default "0:0:0:0:0:FFFF::/96")
  -n, --node-address string           IPv6 address of node, must be in correct format
      --restore                       Restores state, if possible, from previous daemon
      --socket-path string            Sets the socket path to listen for connections (default "/var/run/cilium/cilium.sock")
      --state-dir string              Path to directory to store runtime state (default "/var/run/cilium")
  -t, --tunnel string                 Tunnel mode vxlan or geneve, vxlan is the default (default "vxlan")
      --version                       Print version information
```

**Comment by mellotron** [comment]
Removing the offending "--ipv4" from args returns the following:

```
INFO      _ _ _
INFO  ___|_| |_|_ _ _____
INFO |  _| | | | | |     |
INFO |___|_|_|_|___|_|_|_|
INFO Cilium 0.8.90 3dd63a5 Mon, 24 Apr 2017 14:38:28 +0200 go version go1.8.1 linux/amd64
INFO clang and kernel versions: OK!
INFO linking environment: OK!
INFO BPF Verifier: NOT OK. HAVE_LRU_MAP_TYPE: Your kernel doesn't support LRU maps for BPF, thus switching back to using hash table for the cilium connection tracker. Recommendation is to run 4.10+ kernels.
INFO Using local storage for key-value store
INFO Valid label prefix configuration:
INFO  - :id.
INFO  - :io.cilium.
INFO  - k8s:io.kubernetes.pod.namespace
INFO Using local storage as key-value store
DEBU Creating 8 worker threads
CRIT Error while creating daemon: Error loading config file "/var/lib/kubelet/kubeconfig": read /var/lib/kubelet/kubeconfig: is a directory
```

**Comment by tgraf** [comment]
I'm working on a new `DaemonSet` file here:
https://github.com/cilium/cilium/blob/k8s-doc/examples/minikube/cilium-ds.yaml

You basically just have to make two changes:
- Point to the k8s API server with `--k8s-api-serve`
- Point to either consul or etcd, e.g. `--etcd http://10.1.1.1:2379/`

**Comment by tgraf** [comment]
Fixed in #1020
